runner/context.ts

```diff
@@ -89,74 +94,69 @@ export class Context extends events.EventEmitter {
    * @return {!Context}
    */
   emitHook(name: 'prepare:webserver',
-           app: Express.Application, done: (err?: any) => void): Context;
-  emitHook(name: 'configure', done: (err?: any) => void): Context;
-  emitHook(name: 'prepare', done: (err?: any) => void): Context;
-  emitHook(name: 'cleanup', done: (err?: any) => void): Context;
-  emitHook(name: string, done: (err?: any) => void): Context;
-  emitHook(name: string, ...args: any[]): Context;
-  emitHook(name: string, done: (err?: any) => void): Context {
-    done = done || ((e) => {});
+           app: Express.Application, done: (err?: any) => void): Promise<void>;
+  emitHook(name: 'configure', done: (err?: any) => void): Promise<void>;
+  emitHook(name: 'prepare', done: (err?: any) => void): Promise<void>;
+  emitHook(name: 'cleanup', done: (err?: any) => void): Promise<void>;
+  emitHook(name: string, done: (err?: any) => void): Promise<void>;
+  emitHook(name: string, ...args: any[]): Promise<void>;
+  async emitHook(name: string, done: (err?: any) => void): Promise<void> {
     this.emit('log:debug', 'hook:', name);
 
     const hooks = (this._hookHandlers[name] || []);
-    let boundHooks: ((cb: (err: any) => void) => void)[];
+    let boundHooks: ((cb: (err: any) => void) => (void|Promise<any>))[];
     if (arguments.length > 2) {
-      const hookArgs = Array.from(arguments).slice(1, arguments.length - 1);
       done = arguments[arguments.length - 1];
+      let argsEnd = arguments.length - 1;
+      if (!(done instanceof Function)) {
+        done = (e) => {};
+        argsEnd = arguments.length;
+      }
+      const hookArgs = Array.from(arguments).slice(1, argsEnd);
       boundHooks = hooks.map(function(hook) {
         return hook.bind.apply(hook, [null].concat(hookArgs));
       });
     }
     if (!boundHooks) {
       boundHooks = <any>hooks;
     }
+    done = done || ((e) => {});
 
     // We execute the handlers _sequentially_. This may be slower, but it gives us
     // a lighter cognitive load and more obvious logs.
     let promise = Promise.resolve(null);
     for (const hook of boundHooks) {
       promise = promise.then(() => {
         return new Promise((resolve, reject) => {
-          hook((err) => {
+          const maybePromise = hook((err) => {
```

Comment:
since you're in an async function, rather than chain the promises, you can just await them.